Harvest.Core/Data/DbInitializer.cs

```diff
@@ -15,9 +16,32 @@ public DbInitializer(AppDbContext dbContext)
             _dbContext = dbContext;
         }
 
-        public Task Initialize()
+        public async Task Initialize(bool recreateDb)
         {
-            return Task.CompletedTask;
+            if (recreateDb)
+            {
+                //do what needs to be done?
+            }
+
+            //Make sure roles exist
+            await CheckCreateRole("System");
+            await CheckCreateRole("Admin");
+            await CheckCreateRole("Supervisor");
+            await CheckCreateRole("Worker");
+
+            return;
+        }
+
+        private async Task CheckCreateRole(string role)
+        {
+            if (!await _dbContext.Roles.AnyAsync(a => a.Name == role))
+            {
+                var roleToCreate = new Role {Name = role};
+                await _dbContext.Roles.AddAsync(roleToCreate);
+                await _dbContext.SaveChangesAsync();
+            }
+            
+            return;
```

Comment:
@jSylvestre suggestion: do the SaveChangesAsync() outside of this method.  No reason to do multiple saves when you can batch saves together, and it'll only get more important as the init method grows